### PR TITLE
refactor(stickiness): refactor one stickiness test to use api

### PR DESCRIPTION
**Pull request by hazzadous**
This change demonstrates how to migrate an insights `Query` object level test to
an api based test. It purely focuses on the method of action invocation
and not on any of the e.g. setup or assertions. The StickinessQuery
object is only used by the REST API (and benchmarking), where as the
REST stickiness API is used by external users including our own frontend
developers, so makes sense to test at this level.

## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

## How did you test this code?

*Please describe.*

### CODE REVIEW

File: posthog/api/utils.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def retrieve_entity_from(entity_id: Union[str, int], events: List[Dict], actions
         if event.get("id") == entity_id:
             return event
     for action in actions:
-        if action.get("id") == entity_id:
+        if str(action.get("id")) == entity_id:
```
Comment (macobo) [inline posthog/api/utils.py:49]: Q: Why was this change needed? It will create a conflict with https://github.com/PostHog/posthog/pull/7298

**Review by macobo — COMMENTED** [review]
Small note/wtf on the util, lgtm otherwise.

File: posthog/api/utils.py (hunk 1 above)
Comment (hazzadous) [reply posthog/api/utils.py:59]: @macobo I've updated with a [little comment](https://github.com/PostHog/posthog/pull/7398/commits/e67535174db6f0a32faa033b02f7055e94d27c2b) around the change, and added more restrictive typing

Comment (hazzadous) [reply posthog/api/utils.py:59]: @macobo actually, probably best to get this in now, as the conflicts will be easier to resolve the other way e.g. that way it's just some files should be removed.
